operators-examples/qiskit-dev-operator/operator/pkg/controller/qiskitplayground/qiskitplayground_controller.go

```diff
@@ -3,26 +3,252 @@ package qiskitplayground
+// blank assignment to verify that ReconcileQiskitPlayground implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileQiskitPlayground{}
+
+// ReconcileQiskitPlayground reconciles a QiskitPlayground object
+type ReconcileQiskitPlayground struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+const port = 8888
+const image = "singhp11/centos-jupyter"
+
+func deploymentName(q *singhp11v1.QiskitPlayground) string {
+	return q.Name
+}
+
+func serviceName(q *singhp11v1.QiskitPlayground) string {
+	return q.Name
+}
+
+func routeName(q *singhp11v1.QiskitPlayground) string {
+	return q.Name
+}
+
+func (r *ReconcileQiskitPlayground) ensureDeployment(request reconcile.Request, instance *singhp11v1.QiskitPlayground, dep *appsv1.Deployment,) (*reconcile.Result, error) {
+	// See if deployment already exists and create if it doesn't
+	found := &appsv1.Deployment{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{
+		Name:      dep.Name,
+		Namespace: instance.Namespace,
+	}, found)
+	if err != nil && errors.IsNotFound(err) {
+
+		// Create the deployment
+		log.Info("Creating a new Deployment", "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
+		err = r.client.Create(context.TODO(), dep)
+
+		if err != nil {
+			// Deployment failed
+			log.Error(err, "Failed to create new Deployment", "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
+			return &reconcile.Result{}, err
+		} else {
+			// Deployment was successful
+			return nil, nil
+		}
+	} else if err != nil {
+		// Error that isn't due to the deployment not existing
+		log.Error(err, "Failed to get Deployment")
+		return &reconcile.Result{}, err
+	}
+
+	return nil, nil
+}
+
+func (r *ReconcileQiskitPlayground) ensureService(request reconcile.Request, instance *singhp11v1.QiskitPlayground, s *corev1.Service,) (*reconcile.Result, error) {
+	found := &corev1.Service{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{
+		Name:      s.Name,
+		Namespace: instance.Namespace,
+	}, found)
+	if err != nil && errors.IsNotFound(err) {
+
+		// Create the service
+		log.Info("Creating a new Service", "Service.Namespace", s.Namespace, "Service.Name", s.Name)
+		err = r.client.Create(context.TODO(), s)
+
+		if err != nil {
+			// Creation failed
+			log.Error(err, "Failed to create new Service", "Service.Namespace", s.Namespace, "Service.Name", s.Name)
+			return &reconcile.Result{}, err
+		} else {
+			// Creation was successful
+			return nil, nil
+		}
+	} else if err != nil {
+		// Error that isn't due to the service not existing
+		log.Error(err, "Failed to get Service")
+		return &reconcile.Result{}, err
+	}
+
+	return nil, nil
+}
+
+func (r *ReconcileQiskitPlayground) ensureRoute(request reconcile.Request, instance *singhp11v1.QiskitPlayground, rte *routev1.Route,) (*reconcile.Result, error) {
```

Comment:
Can you create a generic "check-and-if-not-exists-then-create" generic method that takes the type of object as an arg?  Eg.
```
func func (r *ReconcileQiskitPlayground) createIfMissing(kind string, request reconcile.Request, instance *singhp11v1.QiskitPlayground, obj  *Object /*or something generic like this*/) {
...
}